tubetool/nodes/autocrafter.lua

```diff
@@ -0,0 +1,66 @@
+--
+-- Register autocrafter for tubetool
+--
+
+local definition = {
+	name = 'autocrafter',
+	nodes = {
+		"pipeworks:autocrafter",
+	},
+	group = 'autocrafter',
+	protection_bypass_read = "interact",
+}
+
+function definition:copy(node, pos, player)
+	local meta = minetest.get_meta(pos)
+	local inv = meta:get_inventory()
+
+	-- get and store recipe inventory data
+	local recipe = {}
+	for slot = 1, inv:get_size("recipe") do
+		local stack = inv:get_stack("recipe", slot)
+		table.insert(recipe, stack:get_name())
+	end
+
+	-- return data required for replicating this autocrafter configuration
+	return {
+		description = meta:get_string("infotext"),
+		recipe = recipe,
+		enabled = meta:get_int("enabled"),
+	}
+end
+
+function definition:paste(node, pos, player, data)
+	if #data.recipe < 2 then
+		return false
+	end
```

Comment:
This check probably is not needed for anything, also checking if recipe size is less than 2 also does not make much sense...